### ISSUE TITLE
names_containing_dash is broken and unignorable

### ISSUE THREAD

**Issue by leoarnold** [rodjek/puppet-lint#534]
Ruby 2.3.1 / Puppet-Lint 2.0.1

Minimum failing example:

```Puppet
# test.pp
$variable = 'value'
$this_should_not_trigger = "${variable}-something"
$this_should_not_cause_anything = "${variable}-something" # lint:ignore:names_containing_dash
```

causes

```shell
$ bundle exec puppet-lint --show-ignored test.pp 
WARNING: variable contains a dash on line 2
WARNING: variable contains a dash on line 3
```

**Comment by rnelson0** [comment]
@leoarnold We released v2.0.2 to fix this as the rspec tests did not catch this regression. Now, the check will work properly again but the `--fix` will not. We need to revisit the tests before attempting to fix this again, which will be tracked in #504. Thanks.
